omod/src/main/java/org/openmrs/module/datafilter/web/DataFilterWebComponentRegistrar.java

```diff
@@ -17,22 +17,27 @@
 
 import javax.servlet.FilterRegistration;
 import javax.servlet.ServletContext;
+import javax.servlet.ServletContextEvent;
+import javax.servlet.ServletContextListener;
 
 import org.springframework.stereotype.Component;
-import org.springframework.web.context.ServletContextAware;
 
 /**
  * Registers web components that typically are registered via web.xml e.g. filters, servlets and
  * listeners.
  */
 @Component
-public class DataFilterWebComponentRegistrar implements ServletContextAware {
-	
+public class DataFilterWebComponentRegistrar implements ServletContextListener {
 	@Override
-	public void setServletContext(ServletContext servletContext) {
+	public void contextInitialized(ServletContextEvent servletContextEvent) {
+		ServletContext servletContext = servletContextEvent.getServletContext();
 		FilterRegistration filterReg = servletContext.addFilter("datafilter-Filter", new DataFilterWebFilter());
 		//TODO Should ignore static content requests
 		filterReg.addMappingForUrlPatterns(EnumSet.of(ERROR, FORWARD, REQUEST), true, "/*");
 	}
-	
```

Comment:
why the space?

Reply:
If you referring the `new line` that is because there is a new method added to the file. So there is a blank line between two methods.

Reply:
I don't think this works, contextInitialized() is called before modules are loaded.